Sources/Workspace/DefaultExtensionRunner.swift

```diff
@@ -0,0 +1,189 @@
+/*
+This source file is part of the Swift.org open source project
+
+Copyright (c) 2021 Apple Inc. and the Swift project authors
+Licensed under Apache License v2.0 with Runtime Library Exception
+
+See http://swift.org/LICENSE.txt for license information
+See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Basics
+import TSCBasic
+import TSCUtility
+import PackageGraph
+import PackageLoading   // for ManifestResourceProvider
+import PackageModel
+import SPMBuildCore
+import Foundation
+
+
+/// An extension runner that compiles the extension as an executable binary for the host platform, and invokes it as a subprocess.
+public struct DefaultExtensionRunner: ExtensionRunner {
```

Comment:
I may have missed it, but I did not see any sandboxing here. is that still pending?

Reply:
saw the comment about this coming in following PR

Reply:
Yes, this will need to be added to both the running of this script and the command that gets created.